pkg/cluster/builder.go

```diff
@@ -105,7 +101,49 @@ func (b FileSpecBuilder) Build(clusterConfigURL string) (*Spec, error) {
 	config.Cluster.Spec.EksaVersion = &releaseVersion
 	eksaRelease := buildEKSARelease(release, bundlesManifest)
 
-	return NewSpec(config, bundlesManifest, eksd, eksaRelease)
+	return NewSpec(config, bundlesManifest, eksdReleases, eksaRelease)
+}
+
+func getAllEksdReleases(cluster *v1alpha1.Cluster, bundlesManifest *releasev1.Bundles, reader bundles.Reader) ([]eksdv1.Release, error) {
+	m := make([]eksdv1.Release, 0)
```

Comment:
this will always have at least size 1
```suggestion
	m := make([]eksdv1.Release, 0, 1)
```